src/core/Akka.Remote/RemoteActorRefProvider.cs

```diff
@@ -490,7 +490,7 @@ public IInternalActorRef ResolveActorRefWithLocalAddress(string path, Address lo
                 {
                     if (actorPath is RootActorPath)
                         return RootGuardian;
-                    return _local.ResolveActorRef(RootGuardian, actorPath.ElementsWithUid);
+                    return (IInternalActorRef)ResolveActorRef(path); // so we can use caching
```

Comment:
As it turns out, our LRU `IActorRef` Cache was only being used in a tiny handful of niche scenarios before - i.e. for `MiscMessageSerializer` and `SystemMessageSerializer` use cases.

Reply:
Thing that kind of sucks about this design is that on a cache miss we have to parse the `ActorPath` twice now, but on cache hits we can skip all of the actor hierarchy traversal needed to resolve the child.

Reply:
I could, in theory, just check the first cache without adding the item - we have to have the `HasAddress` check run first.

Reply:
With a bit of work I can probably get rid of the second `ActorPath.Parse` call.

Reply:
Resolved this issue by having the ActorRef cache call the ActorPath cache the second time through